v2/tools/generator/internal/codegen/pipeline/create_arm_types.go

```diff
@@ -375,17 +396,44 @@ func (c *armTypeCreator) createConfigMapReferenceProperty(prop *astmodel.Propert
 	return prop.WithType(newType), nil
 }
 
-func (c *armTypeCreator) createARMProperty(prop *astmodel.PropertyDefinition, _ bool) (*astmodel.PropertyDefinition, error) {
+func (c *armTypeCreator) createARMProperty(
+	prop *astmodel.PropertyDefinition,
+	convContext *armPropertyTypeConversionContext,
+) (*astmodel.PropertyDefinition, error) {
 	newType, err := c.createARMTypeIfNeeded(prop.PropertyType())
 	if err != nil {
 		return nil, err
 	}
-	return prop.WithType(newType), nil
+
+	// Return a property with (potentially) a new type
+	result := prop.WithType(newType)
+
+	removeOmitEmpty := false
```

Comment:
very minor: I'm not sure if this would help readability at all, but would it make sense to put this removing omitEmpty bit into its own function?

happy if you think it won't, I'm on the fence.

Reply:
I don't think it would - but I restructured the code to be clearer another way.